instrumentation/kafka-clients/kafka-clients-0.11/library/src/main/java/io/opentelemetry/instrumentation/kafkaclients/TracingConsumerInterceptor.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kafkaclients;
+
+import java.util.Map;
+import org.apache.kafka.clients.consumer.ConsumerInterceptor;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+
+public class TracingConsumerInterceptor<K, V> implements ConsumerInterceptor<K, V> {
```

Comment:
The opentracing lib you linked also contains an implementation of a tracing `Consumer`/`Producer` decorator: https://github.com/opentracing-contrib/java-kafka-client/blob/master/opentracing-kafka-client/src/main/java/io/opentracing/contrib/kafka/TracingKafkaConsumer.java
Just wondering, why did you decide to use the interceptors instead? They need to be passed as class name (and thus require a no-arg constructor), while you can very easily decorate `Consumer`/`Producer` in plain Java code (Spring Kafka even has cool `ProducerPostProcessor`/`ConsumerPostProcessor` types that allow to wrap the original Kafka type with your own decorator)

Reply:
Also, one advantage of using the decorator is that you'll get the timings right - e.g. you start the span just before the message is sent and end it just after it's sent; with the current approach you'll get practically 0-length spans.

Reply:
Currently I've just added what we were missing in [Strimzi](https://strimzi.io/) -- the interceptors.

I'll check the rest of the OpenTracing Kafka support later, and try to port it over as well.

Reply:
Hmm, so you can't use anything other than interceptors?

Reply:
That's how we use it now ... and to be on the safe side, config-wise, I just ported the same behavior.